From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct pci_dev
Patch-mainline: never, kabi
References: kabi

Upstream commit 4ec73791a64bab25cabf16a6067ee478692e506d (PCI: Work
around Pericom PCIe-to-PCI bridge Retrain Link erratum) added a bit to
struct pci_dev. It made the kABI checker to complain.

Given the bit uses a hole, hide the change from the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/pci.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -328,8 +328,10 @@ struct pci_dev {
 	unsigned int	hotplug_user_indicators:1; /* SlotCtl indicators
 						      controlled exclusively by
 						      user sysfs */
+#ifndef __GENKSYMS__
 	unsigned int	clear_retrain_link:1;	/* Need to clear Retrain Link
 						   bit manually */
+#endif
 	unsigned int	d3_delay;	/* D3->D0 transition time in ms */
 	unsigned int	d3cold_delay;	/* D3cold->D0 transition time in ms */
 
